lib/transporters.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as request from 'request';
+
+// tslint:disable-next-line
+const pkg = require('../package.json');
+
+export default class DefaultTransporter {
+
+  /**
+   * Default user agent.
+   */
+  public static USER_AGENT = 'google-api-nodejs-client/' + pkg.version;
+
+  /**
+   * Configures request options before making a request.
+   * @param {object} opts Options to configure.
+   * @return {object} Configured options.
+   */
+  public configure(opts: any): any {
```

Comment:
It would be nice to provide a type for these options so there is a single place to look for option/config structure.